diff --git a/package/kernel/mt76/Makefile b/package/kernel/mt76/Makefile
--- a/package/kernel/mt76/Makefile
+++ b/package/kernel/mt76/Makefile
@@ -9,7 +9,7 @@
 PKG_SOURCE_URL:=https://github.com/openwrt/mt76
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_DATE:=2026-08-01
-PKG_SOURCE_VERSION:=b2704cf5a4068b672bf47ad5bf6b4802b6770a90
+PKG_SOURCE_DATE:=2026-07-01
+PKG_SOURCE_VERSION:=59676919ea408b0b13a9d23f2e2e1a1ab407fba1
 PKG_MIRROR_HASH:=skip
