diff --git a/package/kernel/mt76/Makefile b/package/kernel/mt76/Makefile
index 3d41a46b8faa37..b95c2f4d40d004 100644
--- a/package/kernel/mt76/Makefile
+++ b/package/kernel/mt76/Makefile
@@ -9,7 +9,7 @@ PKG_LICENSE_FILES:=
 
 PKG_SOURCE_URL:=https://github.com/openwrt/mt76
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_DATE:=2026-07-29
-PKG_SOURCE_VERSION:=7f4a3b710a983981afc813defab0b7bd4a9300ec
+PKG_SOURCE_DATE:=2026-08-01
+PKG_SOURCE_VERSION:=b2704cf5a4068b672bf47ad5bf6b4802b6770a90
 PKG_MIRROR_HASH:=skip
